### PR TITLE
UI - Migrate to Angular 15

### PR DESCRIPTION
**Pull request by ThibaudAV**
## :id: Reference related issue. 
n/a

## :pencil2: A description of the changes proposed in the pull request

See commits list
but mainly :
- Update to angular 15
- Clean angular project
- Update material to mdc & fix UI

📝 (Viewed with the UI/UX team.) The theme from ui-particles was partially applied, so I applied it completely to make it work properly.

## :memo: Test scenarios 

Go through all the pages and dialogs to see if they are displayed correctly

## :computer: Add screenshots for UI
hum Test it locally 🙏 

## :books: Any other comments that will help with documentation
n/a

## :man: :woman:@mentions of the person or team responsible for reviewing proposed changes

This is a nice example: https://github.com/gravitee-io/gravitee-access-management/pull/1822

### CODE REVIEW

File: gravitee-am-ui/package.json [hunk 1]
```diff
@@ -24,86 +25,84 @@
...
+    "@angular/platform-browser": "15.2.10",
+    "@angular/platform-browser-dynamic": "15.2.10",
+    "@angular/platform-server": "15.2.10",
+    "@angular/router": "15.2.10",
     "@gravitee/ui-components": "3.45.2",
-    "@gravitee/ui-particles-angular": "7.58.0",
-    "@ngx-formly/core": "6.1.8",
-    "@ngx-formly/material": "6.1.8",
-    "@swimlane/ngx-datatable": "19.0.0",
+    "@gravitee/ui-particles-angular": "^9.0.2-ui-fix-from-am-2dde43c",
```
Comment (ThibaudAV) [inline gravitee-am-ui/package.json:41]: 📝  wait ui-particles release

**Comment by ThibaudAV** [comment]
I found some new errors in the console. I'm going to try and fix theme, but it doesn't seem to break pages. You can already start testing if you like. 🙏

**Comment by ThibaudAV** [comment]
Don't hesitate to call me for a review, if needed :)

File: gravitee-am-ui/src/app/domain/settings/users/user/devices/devices.component.ts [hunk 2]
```diff
@@ -34,6 +35,7 @@ export class UserDevicesComponent implements OnInit {
   canDelete: boolean;
   devices: any[];
   deviceIdentifiers: any[];
+  sortType = SortType.multi;
```
Comment (leleueri) [inline gravitee-am-ui/src/app/domain/settings/users/user/devices/devices.component.ts:38]: **question**: is it a good practice in UI to do this ? When I read the change in the template, I was thinking there is an error until reading this change :thinking:

File: gravitee-am-ui/src/app/components/navbar/navbar.component.html [hunk 3]
```diff
@@ -21,12 +21,12 @@
       <button mat-icon-button gioTopBarHome>
         <mat-icon svgIcon="gio:gravitee" (click)="navigateToHome()"></mat-icon>
       </button>
-      <gio-top-bar-content type="am" productName="Access Management"></gio-top-bar-content>
+      <gio-top-bar-content type="apim" productName="API Management"></gio-top-bar-content>
```
Comment (leleueri) [inline gravitee-am-ui/src/app/components/navbar/navbar.component.html:24]: **question**: is it intentional ?

Comment (ThibaudAV) [reply gravitee-am-ui/src/app/components/navbar/navbar.component.html:24]: oups nope. error on my part i had a problem with the height and i tested with the APIM one. sorry

File: gravitee-am-ui/src/app/domain/settings/users/user/devices/devices.component.ts (hunk 2 above)
Comment (ThibaudAV) [reply gravitee-am-ui/src/app/domain/settings/users/user/devices/devices.component.ts:38]: This is to use the ngx-datatable enum As typescript is stricter, it indicates this error
image

**Comment by leleueri** [comment]
@ThibaudAV look like the CI is failing on the UI build, Node.js can't be downloaded

**Comment by ThibaudAV** [comment]
ra node & npm inside mvn 🙈

**Review by leleueri — APPROVED** [review]

**Pull request merged**